### PR TITLE
Implement convert_na_to

### LINKED ISSUE #72: Add function to replace `NA`s
We need to avoid collision with [_tidyr_ function](https://tidyr.tidyverse.org/reference/replace_na.html), so need to think of another name than the one used here: https://strengejacke.github.io/sjmisc/reference/replace_na.html

### PR DESCRIPTION
**Pull request by etiennebacher**
This PR is related to #72. It implements `convert_na_to()`, a function to replace NA in vectors and dataframe. 

### Behavior

For numeric and character vectors, the behavior is quite simple:
``` r
# Convert NA to 0 in a numeric vector
datawizard::convert_na_to(
  c(9, 3, NA, 2, 3, 1, NA, 8),
  replacement = 0
)
#> [1] 9 3 0 2 3 1 0 8

# Convert NA to "missing" in a character vector
datawizard::convert_na_to(
  c("a", NA, "d", "z", NA, "t"),
  replacement = "missing"
)
#> [1] "a"       "missing" "d"       "z"       "missing" "t"
```

<sup>Created on 2022-03-17 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

For dataframes, the behavior is a bit different, depending on `select`:

* if `select` is `NULL` or is a vector of variable names, then `replace_*` values are used to convert `NA` in the variables specified:
``` r
test_df <- data.frame(
  x = c(1, 2, NA),
  x2 = c(4, 5, NA),
  y = c("a", "b", NA)
)

# Convert all NA to 0 in numeric variables, and all NA to "missing" in
# character variables
datawizard::convert_na_to(
  test_df,
  replace_num = 0,
  replace_char = "missing"
)
#>   x x2       y
#> 1 1  4       a
#> 2 2  5       b
#> 3 0  0 missing

# Convert a specific variable in the dataframe
datawizard::convert_na_to(
  test_df,
  replace_num = 0,
  replace_char = "missing",
  select = "x"
)
#>   x x2    y
#> 1 1  4    a
#> 2 2  5    b
#> 3 0 NA <NA>
```

<sup>Created on 2022-03-17 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

* if `select` is a named list of variables with specific values, then these values will override `replace_*` for the variables concerned, but `replace_*` will be applied to the others:
``` r
test_df <- data.frame(
  x = c(1, 2, NA),
  x2 = c(4, 5, NA),
  y = c("a", "b", NA)
)

# Convert NA to 1 in variable 'x2' and to 0 in all other numeric
# variables
datawizard::convert_na_to(
  test_df,
  replace_num = 0,
  select = list(x2 = 1)
)
#>   x x2    y
#> 1 1  4    a
#> 2 2  5    b
#> 3 0  1 <NA>
```

<sup>Created on 2022-03-17 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

### About factors

I'm not really comfortable with factors, so I'm really not sure about my implementation. This is why there are no examples or tests for factors. Do you have some ideas for that?

Regarding the name, @DominiqueMakowski proposed `convert_na()`. I prefer `convert_na_to()` to be more complementary with `convert_to_na()`, but I have no problem with changing the name if you prefer.

Close #72

### CODE REVIEW

**Comment by IndrajeetPatil** [comment]
The builds are failing due to an undocumented argument:

```r
❯ checking Rd \usage sections ... WARNING
  Undocumented arguments in documentation object 'convert_na_to'
    ‘...’
```

**Comment by strengejacke** [comment]
I think it should be `convert_to_na()`, because the function does not convert na into non na values.

**Comment by strengejacke** [comment]
Ok was reading the examples wrong

**Comment by etiennebacher** [comment]
> The builds are failing due to an undocumented argument:
> 
> ```r
> ❯ checking Rd \usage sections ... WARNING
>   Undocumented arguments in documentation object 'convert_na_to'
>     ‘...’
> ```

Thanks, I fixed it

**Comment by strengejacke** [comment]
PR looks good to me. While we're on it, maybe you could even copy/paste the code for select-helpers I recently added to `convert_to_na()`?

https://github.com/easystats/datawizard/commit/d56677487f01705ed7873ac941158d7faaf730cd

It's roughly replacing `if (inherits(select, "formula")) { select <- all.vars(select) }` with the new code in that commit. I would not yet do this for `exclude`.

**Comment by IndrajeetPatil** [comment]
>I'm not really comfortable with factors, so I'm really not sure about my implementation. This is why there are no examples or tests for factors. Do you have some ideas for that? 

Any thoughts, @strengejacke?

**Comment by etiennebacher** [comment]
> It's roughly replacing `if (inherits(select, "formula")) { select <- all.vars(select) }` with the new code in that commit. I would not yet do this for `exclude`.

Does this mean that formulas are no longer accepted in `select`?

**Comment by strengejacke** [comment]
> > It's roughly replacing `if (inherits(select, "formula")) { select <- all.vars(select) }` with the new code in that commit. I would not yet do this for `exclude`.
> 
> Does this mean that formulas are no longer accepted in `select`?

Good point, usually yes. But my last commit in this PR: https://github.com/easystats/datawizard/pull/114 fixed this. Maybe we can use that?

``` r
datawizard::data_addprefix(iris, "NEW_", select = ~Species + Petal.Width) |> head()
#>   Sepal.Length Sepal.Width Petal.Length NEW_Petal.Width NEW_Species
#> 1          5.1         3.5          1.4             0.2      setosa
#> 2          4.9         3.0          1.4             0.2      setosa
#> 3          4.7         3.2          1.3             0.2      setosa
#> 4          4.6         3.1          1.5             0.2      setosa
#> 5          5.0         3.6          1.4             0.2      setosa
#> 6          5.4         3.9          1.7             0.4      setosa
```

<sup>Created on 2022-03-18 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Comment by strengejacke** [comment]
so we should wait for my PR to be merged.

**Comment by etiennebacher** [comment]
Code of `convert_na_to.dataframe` a bit messy. The thing is that `is.list(select)` can't run too early in the function because if `select = starts_with(...)` it will error "function starts_with doesn't exist".

**Comment by strengejacke** [comment]
I tried to simplify a bit, I guess a list is OK, you just should not pass it to `.evaluated_pattern_to_colnames()`. See my last commit.

File: R/convert_na_to.R
Comment (etiennebacher) [reply R/convert_na_to.R:194]: I think it's just because I forgot to use it. It could still be useful to have `exclude` when `select` is a list

Comment (etiennebacher) [reply R/convert_na_to.R:194]: `exclude` could also have some NSE to use `starts_with()` etc. I feel like the code for `convert_na_to.dataframe` starts being quite messy with all this NSE / list cases, maybe I should wait for #115 to be finished before refactoring `convert_na_to`.

Comment (strengejacke) [reply R/convert_na_to.R:194]: yeah, sounds good.

**Comment by strengejacke** [comment]
ok, once checks pass, this is ready to merge, I think. We should mention this in the news, as well.

**Review by IndrajeetPatil — APPROVED** [review]
Thanks, @etiennebacher, once again for a high quality PR!

I trust Daniel with when this is ready to merge.

**Comment by etiennebacher** [comment]
Thank you! I also learned a lot by watching the development of the NSE function at the same time. I just wanted to say that although there is `convert_na_to.factor`, it is not tested (I commented the lines with factors in the tests) and I'm unsure about how levels are treated. You may wanna check that

**Comment by strengejacke** [comment]
I think the implemented treatment of factors is ok. I uncommented the tests.

**Comment by DominiqueMakowski** [comment]
@etiennebacher thanks, and don't hesitate to contribute or get involved with other things if you're interested ☺️

**Pull request merged**
